types/json_size.go

```diff
@@ -0,0 +1,446 @@
+// json.Marshal with the given value.
+// Since JSON marshalling does not have a guaranteed output format,
+// this should be understood as a best guess and correct in most cases.
+// Do not use it when a precise value is required.
+func (t IBCOpenConfirm) ExpectedJSONSize() int {
+	// Channel IBCChannel `json:"channel"`
+	return brackets + 9 + colon + t.Channel.ExpectedJSONSize()
+}
+
+// ExpectedJSONSize returns the expected JSON size in bytes when using
+// json.Marshal with the given value.
+// Since JSON marshalling does not have a guaranteed output format,
+// this should be understood as a best guess and correct in most cases.
+// Do not use it when a precise value is required.
+func (t IBCChannelConnectMsg) ExpectedJSONSize() int {
+	// OpenAck     *IBCOpenAck     `json:"open_ack,omitempty"`
+	// OpenConfirm *IBCOpenConfirm `json:"open_confirm,omitempty"`
+	out := brackets
+	if t.OpenAck != nil {
+		out += 10 + colon + t.OpenAck.ExpectedJSONSize()
+	}
+	if t.OpenAck != nil && t.OpenConfirm != nil {
+		// this case should not happen for an enum but we don't have an error return type, so just handle it
+		out += comma
+	}
+	if t.OpenConfirm != nil {
+		out += 14 + colon + t.OpenConfirm.ExpectedJSONSize()
+	}
+	return out
+}
+
+// ExpectedJSONSize returns the expected JSON size in bytes when using
+// json.Marshal with the given value.
+// Since JSON marshalling does not have a guaranteed output format,
+// this should be understood as a best guess and correct in most cases.
+// Do not use it when a precise value is required.
+func (t IBCCloseInit) ExpectedJSONSize() int {
+	// Channel IBCChannel `json:"channel"`
+	return brackets + 9 + colon + t.Channel.ExpectedJSONSize()
+}
+
+// ExpectedJSONSize returns the expected JSON size in bytes when using
+// json.Marshal with the given value.
+// Since JSON marshalling does not have a guaranteed output format,
+// this should be understood as a best guess and correct in most cases.
+// Do not use it when a precise value is required.
+func (t IBCCloseConfirm) ExpectedJSONSize() int {
+	// Channel IBCChannel `json:"channel"`
+	return brackets + 9 + colon + t.Channel.ExpectedJSONSize()
+}
+
+// ExpectedJSONSize returns the expected JSON size in bytes when using
+// json.Marshal with the given value.
+// Since JSON marshalling does not have a guaranteed output format,
+// this should be understood as a best guess and correct in most cases.
+// Do not use it when a precise value is required.
+func (t IBCChannelCloseMsg) ExpectedJSONSize() int {
+	// CloseInit    *IBCCloseInit    `json:"close_init,omitempty"`
+	// CloseConfirm *IBCCloseConfirm `json:"close_confirm,omitempty"`
+	out := brackets
+	if t.CloseInit != nil {
+		out += 12 + colon + t.CloseInit.ExpectedJSONSize()
+	}
+	if t.CloseInit != nil && t.CloseConfirm != nil {
+		// this case should not happen for an enum but we don't have an error return type, so just handle it
+		out += comma
```

Comment:
nit: Wouldn't this case have nothing except the brackets anyways? Why the comma?

Reply:
Ah, you are totally correct.